File: frontend/src/types/assessment.ts
export type AbilityScores = {
  logic: number;
  expression: number;
  spatialDesign: number;
  interpersonal: number;
};

export type InterestScores = {
  handsOn: number;
  research: number;
  creation: number;
  helping: number;
  leadership: number;
  detail: number;
};

export type AssessmentResponseInput = {
  grade: string;
  gender: string;
  collegeMajor: string;
  hometown?: string;
  mastersIntention: string;
  mastersPlan?: string;
  phdIntention: string;
  phdPlan?: string;
  educationPathReasons: string[];
  educationCertainty: number;
  fiveYearCity: string;
  fiveYearIncome: string;
  fiveYearIndustry: string;
  fiveYearRole: string;
  fiveYearFamilyStatus: string;
  fiveYearHousingPlan: string;
  fiveYearHobbiesSkills: string;
  tenYearCity: string;
  tenYearIncome: string;
  tenYearIndustry: string;
  tenYearRole: string;
  tenYearFamilyStatus: string;
  tenYearHousingPlan: string;
  tenYearHobbiesSkills: string;
  topValuesRanked: string[];
  abilityScores: AbilityScores;
  interestScores: InterestScores;
  currentPreparations: string[];
  missingResources: string[];
  majorOutcomeAwareness: string;
  targetJobAwareness: string;
  jobInfoChannels: string[];
  healthEnergyStatus: string;
  exerciseFrequency?: string;
  careerConfusions: string[];
  mainConfusionText?: string;
};

File: backend/app/schemas/assessment.py
from __future__ import annotations

from typing import Optional

from pydantic import BaseModel, Field


class AbilityScores(BaseModel):
    logic: int = Field(ge=1, le=5)
    expression: int = Field(ge=1, le=5)
    spatialDesign: int = Field(ge=1, le=5)
    interpersonal: int = Field(ge=1, le=5)


class InterestScores(BaseModel):
    handsOn: int = Field(ge=1, le=5)
    research: int = Field(ge=1, le=5)
    creation: int = Field(ge=1, le=5)
    helping: int = Field(ge=1, le=5)
    leadership: int = Field(ge=1, le=5)
    detail: int = Field(ge=1, le=5)


class AssessmentResponseInput(BaseModel):
    grade: str
    gender: str
    collegeMajor: str
    hometown: Optional[str] = None
    mastersIntention: str
    mastersPlan: Optional[str] = None
    phdIntention: str
    phdPlan: Optional[str] = None
    educationPathReasons: list[str]
    educationCertainty: int = Field(ge=1, le=5)
    fiveYearCity: str
    fiveYearIncome: str
    fiveYearIndustry: str
    fiveYearRole: str
    fiveYearFamilyStatus: str
    fiveYearHousingPlan: str
    fiveYearHobbiesSkills: str
    tenYearCity: str
    tenYearIncome: str
    tenYearIndustry: str
    tenYearRole: str
    tenYearFamilyStatus: str
    tenYearHousingPlan: str
    tenYearHobbiesSkills: str
    topValuesRanked: list[str]
    abilityScores: AbilityScores
    interestScores: InterestScores
    currentPreparations: list[str]
    missingResources: list[str]
    majorOutcomeAwareness: str
    targetJobAwareness: str
    jobInfoChannels: list[str]
    healthEnergyStatus: str
    exerciseFrequency: Optional[str] = None
    careerConfusions: list[str]
    mainConfusionText: Optional[str] = None
    userId: Optional[str] = None


class AssessmentResponse(AssessmentResponseInput):
    id: str
    userId: str
    submittedAt: str
    createdAt: str


class AssessmentSubmitResult(BaseModel):
    userId: str
    responseId: str
    profileId: str
    reportId: str
    generationStatus: str
